### ISSUE TITLE
Be able to set elements in dynamic forms as readonly for certain users

### ISSUE THREAD

**Issue by jersiovic** [OrchardCMS/Orchard#6835]
It would be interesting to have for FormElements a ReadOnlyRule similar to VisibilityRule that allow to decide dynamically using Rules if an element can be editable or not.

**Comment by jersiovic** [comment]
Done in XKProject repo. 
Here you have differences with https://github.com/OrchardCMS/Orchard/issues/6816 issue (Ajax support for taxonomy element to have a chain of select lists each showing different levels of a hierarchy) https://github.com/xkproject/Orchard/compare/%236816_TaxonomyElementWithAjax...xkproject:%236835_ReadonlyRuleFormElements

I cannot send https://github.com/OrchardCMS/Orchard/issues/6816 issue as PR cause that other Issue is not clear if it is ok as it is for merging it in core. So, this ReadOnlyRule issue till previous one is not clear cannot be submitted as PR.

**Comment by jersiovic** [comment]
Added also in this branch support for tokens in VisibilityRule and in ReadOnlyRule
Added Token Element.GetContentToEdit that provides access to contentitem that is going to be edited
Finally also added conditions CompareCondition and IsNullOrWhiteSpaceCondition.

This allows to evaluate data of the edited content in Visibility Rule and ReadOnly Element. An example is this exppression I use in a Row Element within a Form to decide Visibility:
```
compare('{Element.ContentToEdit.Fields.BoatPart.PublishingState}','Rejected')))
```

**Comment by sebastienros** [comment]
Not sure it's easily doable as each form element would need to support this, contrary to VisibleRule. I will wait for other users to comment on the necessity.

**Comment by sfmskywalker** [comment]
Instead of implementing it similar to the VisibilityRule, you could implement it as an additional property for all elements that derive from `FormElement` so that the property only applies to Dynamic Form elements.

**Comment by jersiovic** [comment]
@sfmskywalker yes I did it as you comment: xkproject@ #6816_TaxonomyElementWithAjax...xkproject:#6835_ReadonlyRuleFormElements 
Concretly this is the FormElement class with the ReadOnlyRule added

**Comment by jersiovic** [comment]
@sebastienros the necessity depends directly on this other issue https://github.com/OrchardCMS/Orchard/issues/6163 . If you implement the dynamic forms for edit content feature then after a while you start to need this other feature.

Related to dynamic forms for edit content feature currently when I have time I'm improving its code to make it more readable. I wiil ping you when it is ready, in order to hear your opinion and evaluate if a PR could be interesting.
